File: Domain/Master.cs
using System;

namespace Domain
{
    public class Master
    {
        public int Id { get; set; }
        public string FirstName { get; set;}
        public string LastName { get; set;}
        public string FullName { get => FirstName + " " + LastName; }
        public string BirthPlace { get; set;}
        public DateTime? BirthDate { get; set;}
        public string BirthDateFormatted { get => BirthDate.HasValue ? BirthDate.Value.ToString("MMMM d, yyyy") : "No Birth Date Added"; }
        public DateTime? DeathDate { get; set;}
        public string DeathDateFormatted { get => DeathDate.HasValue ? DeathDate.Value.ToString("MMMM d, yyyy") : "No Death Date Added"; }
        public string Bio { get; set; }
        public string Photo { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set;}
    }
}

File: client-app/src/app/models/master.ts
export interface IMaster {
    id: string;
    firstName: string;
    lastName: string;
    fullName?: string;
    birthPlace: string;
    birthDate: Date;
    birthDateFormatted: string;
    deathDate: Date;
    deathDateFormatted: string;
    bio: string;
    photo: string;
    isActive: boolean;
}

export interface IMasterFormValues extends Partial<IMaster> {
    time?: Date;
}

export class MasterFormValues implements IMasterFormValues {
    id?: string = undefined;
    firstName: string = '';
    lastName: string = '';
    birthPlace: string = '';
    birthDate?: Date = undefined;
    deathDate?: Date = undefined;
    bio: string = '';
    photo: string = '';
    isActive: boolean = false;

    constructor(init?: IMasterFormValues) {
        // if (init && init.birthDate) {
        //     init.time = init.birthDate;
        // }  
        Object.assign(this, init);
    }
}